### PR TITLE
Makes the names logged by TracingWorkflowInterceptor human readable.

### PR DESCRIPTION
**Pull request by rjrjr**
Also:

 * Eliminates accidental dependency on kotlin-reflect

 * Drops use of SimpleLoggingWorkflowInterceptor in the samples,
   we don't want it cargo culted

Screen Shot 2021-03-02 at 4 54 09 PM

### CODE REVIEW

**Comment by rjrjr** [comment]
It would be wise to ignore whitespace changes while reviewing this.

**Comment by zach-klippenstein** [comment]
Unfortunately that is not something the mobile app seems to know how to do 😒

File: workflow-tracing/src/main/java/com/squareup/workflow1/diagnostic/tracing/TracingWorkflowInterceptor.kt [hunk 1]
```diff
@@ -481,15 +484,46 @@ public class TracingWorkflowInterceptor internal constructor(
...
+
+  val params = arguments.map { projection ->
+    when (val type = projection.type) {
+      is KType -> type.toLoggingName()
+      else -> "*"
+    }
+  }
+
+  return if (params.isEmpty()) classifierName else "$classifierName<${params.joinToString(", ")}>"
+}
```
Comment (zach-klippenstein) [inline workflow-tracing/src/main/java/com/squareup/workflow1/diagnostic/tracing/TracingWorkflowInterceptor.kt:522]: Who needs the reflection library when you can write it yourself? 😅

**Review by zach-klippenstein — APPROVED** [review]

File: workflow-tracing/src/main/java/com/squareup/workflow1/diagnostic/tracing/TracingWorkflowInterceptor.kt (hunk 1 above)
Comment (rjrjr) [reply workflow-tracing/src/main/java/com/squareup/workflow1/diagnostic/tracing/TracingWorkflowInterceptor.kt:522]: I love extension methods so much. Nothing but syntactic sugar, but they make such a huge difference.

**Pull request merged**